### PR TITLE
Log pin actions to audit log

### PR DESCRIPTION
**Pull request by mpolden**

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/ContactInformationMaintainer.java [hunk 1]
```diff
@@ -36,10 +36,13 @@ public ContactInformationMaintainer(Controller controller, Duration interval, Jo
     @Override
     protected void maintain() {
         for (Tenant tenant : controller().tenants().asList()) {
-            try{
-            Optional<PropertyId> tenantPropertyId = tenant instanceof AthenzTenant ? ((AthenzTenant) tenant).propertyId() : Optional.empty();
-            Contact contact = contactRetriever.getContact(tenantPropertyId);
-            controller().tenants().lockIfPresent(tenant.name(), lockedTenant -> controller().tenants().store(lockedTenant.with(contact)));
+            try {
+                Optional<PropertyId> tenantPropertyId = Optional.empty();
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/ContactInformationMaintainer.java]: I honestly read the old version more easily, but I won't object :)

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java [hunk 2]
```diff
@@ -206,8 +206,8 @@ private HttpResponse handlePOST(HttpRequest request) {
...
         if (path.matches("/application/v4/tenant/{tenant}/application/{application}/promote")) return promoteApplication(path.get("tenant"), path.get("application"), request);
-        if (path.matches("/application/v4/tenant/{tenant}/application/{application}/deploying/platform")) return deployPlatform(path.get("tenant"), path.get("application"), readToString(request.getData()), false);
-        if (path.matches("/application/v4/tenant/{tenant}/application/{application}/deploying/pin")) return deployPlatform(path.get("tenant"), path.get("application"), readToString(request.getData()), true);
+        if (path.matches("/application/v4/tenant/{tenant}/application/{application}/deploying/platform")) return deployPlatform(path.get("tenant"), path.get("application"), false, request);
+        if (path.matches("/application/v4/tenant/{tenant}/application/{application}/deploying/pin")) return deployPlatform(path.get("tenant"), path.get("application"), true, request);
         if (path.matches("/application/v4/tenant/{tenant}/application/{application}/deploying/application")) return deployApplication(path.get("tenant"), path.get("application"));
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java]: Probably something to include for this path as well?

**Review by jonmv — APPROVED** [review]
Perhaps wanted one more place, as pointed out. Otherwise, LGTM :)

**Comment by jonmv** [comment]
Unicorn

**Comment by mpolden** [comment]
Java build passed, merging.

**Pull request merged**